### PR TITLE
polymorphic button types

### PR DESCRIPTION
**Pull request by danethurber**
currently the button doesn't support undeclared properties that should be allow. example - the Button should allow a `type` props. 

Screen Shot 2020-10-29 at 3 45 34 PM

i've attempted to update the prop typings to allow button or anchor props....but i can't figure it out. anybody else feel like taking a shot?

### CODE REVIEW

**Comment by jaketrent** [comment]
I'd love to give it a shot.

File: packages/button/src/react/index.tsx [hunk 1]
```diff
@@ -126,89 +129,115 @@ interface ButtonStatics {
...
+  iconAlign?: ValueOf<typeof vars.iconAligns>
   loading?: boolean
-  size?: string
+  size?: ValueOf<typeof vars.sizes>
+}
+
+type AnchorElProps = ButtonProps<HTMLAnchorElement> & { href: string }
+type ButtonElProps = ButtonProps<HTMLButtonElement> & {
+  name?: string
+  type?: string
```
Comment (danethurber) [inline packages/button/src/react/index.tsx:145]: needed to redeclare these really defeats the purpose....

**Comment by jaketrent** [comment]
@danethurber what do you think of the latest commit.  I used Will's approach.  Seems to tsc ok, test ok, and get code completion ok.  I'm probably missing something. 🤷

**Review by danethurber — COMMENTED** [review]
let's try it. i honestly don't know what is correct at this point

**Comment by jaketrent** [comment]
+1

**Review by jaketrent — APPROVED** [review]

**Pull request merged**